### DISCUSSION TITLE
Is it possible to use Hyperpipe as music provider

### DISCUSSION THREAD

**Discussion by SaharaMiyoko** [KRTirtho/spotube#898] [Ideas]
I found that Hyperpipe can get music from YT Music directly, so I'm wondering whether Spotube can use Hyperpipe as a music provider?(I'm busy with my works, so I think I don't have time to check out my Github messages)

**Comment by KRTirtho** [comment]
I guess it uses piped api which what currently Spotube also uses. So its literally the same thing

**Reply by SaharaMiyoko** [reply]
<img width="953" alt="屏幕截图 2023-11-26 091757" src="https://github.com/KRTirtho/spotube/assets/104242390/9689abb2-9393-4730-b03d-bd229241054c">
No, it uses not only piped api but hyperpipe api, look at this picture. And I noticed that some of the songs weren't playing directly from the MV on YouTube. Then I went to look at the repository for the project(https://codeberg.org/Hyperpipe/Hyperpipe), so I guessed that this api was getting the music from Youtube Music, not Youtube.

**Reply by KRTirtho** [reply]
Hyperpipe backend is just for getting YT Music playlist genres etc as NewPipe doesn't support it yet.

**Reply by KRTirtho** [reply]
The actual audio source and search etc. are still using Piped
